In [2]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd


In [25]:
path_to_data = './interactive_data.csv'

data = pd.read_csv(path_to_data, header=0)
data = data.iloc[:, 1:]
data.head()

Intent         Gender            Age                    Race  \
0  None selected  None selected  None selected           None selected   
1  None selected  None selected  None selected                   White   
2  None selected  None selected  None selected                   Black   
3  None selected  None selected  None selected                Hispanic   
4  None selected  None selected  None selected  Asian/Pacific Islander   

   Deaths  Population  Rate  
0   33599   316299978  10.6  
1   22079   197369634  11.2  
2    7765    38896382  20.0  
3    3007    54049078   5.6  
4     442    16315561   2.7

In [28]:
"""This analysis is without filtering"""
unique_intent = data['Intent'].unique() #to get the unique values in that column

suicides = data[data['Intent'] == 'Suicide'] #yeb2a data[----] 3shan teaccess eldataframe, w gowaha t7ot elcondition, w fel condition 3andak data['Intent'] == 'Suicide', 3shan yeb2a m3ak elcondition
suicides_number = sum(suicides['Deaths'])
total_deaths = sum(data['Deaths'])
suicides_ratio = suicides_number/total_deaths 
print(suicides_ratio)

['None selected' 'Suicide' 'Homicide' 'Accident' 'Unknown']
0.3133747535250567


In [33]:
"""This analysis is done wuth the filtering"""

#1st, we filter the dataframe
value_to_drop = 'None selected'
filtered_data = data[~(data == value_to_drop).any(axis=1)]

#Then, we operate the analysis on the suicide/deaths ratio.
filtered_suicides = filtered_data[filtered_data['Intent'] == 'Suicide']
filtered_suicides_number = sum(filtered_suicides['Deaths'])
filtered_total_deaths = sum(filtered_data['Deaths'])
filtered_suicide_ratio = filtered_suicides_number / filtered_total_deaths
print(filtered_suicide_ratio)

0.6268194671826165


With a filtered_suicide_ratio = 62.68% ~ 2/3, *we have concluded the 1st assignment*

In [38]:
#Getting the number of deaths by male suicides
male_suicides = (filtered_suicides[filtered_suicides['Gender'] == 'Male'])
male_suicides_count = sum(male_suicides['Deaths'])

#Getting the total number of MALE deaths by suicide
total_suicides_count = sum(filtered_suicides['Deaths'])
male_suicides_ratio = male_suicides_count/total_suicides_count
print(male_suicides_ratio)

0.8624275809668535


So we get that 86% of the Suicides are from MALES. Thus concluding the second assignment.

In [40]:
homicides = filtered_data[filtered_data['Intent'] == 'Homicide']
homicides_count = sum(homicides['Deaths'])
homicides_ratio = homicides_count / filtered_total_deaths
print(homicides_ratio)

0.34906980205387705


Third assignment is DONE.

In [46]:
male_homicides = homicides[homicides['Gender'] == 'Male']
male_homi_18_34 = male_homicides[male_homicides['Age'] == '15 - 34']
male_homi_18_34_ratio = sum(male_homi_18_34['Deaths']) / sum(male_homicides['Deaths'])
print(male_homi_18_34_ratio)

0.6564324541977049


Fourth task is done.

In [48]:
women_homicides = homicides[homicides['Gender'] == 'Female']
women_homicides_ratio = sum(women_homicides['Deaths'])/ homicides_count

print(women_homicides_ratio)

0.15289502856655582


Fifth task is done.